configure.ac

```diff
@@ -95,8 +95,14 @@ AC_ARG_ENABLE([debug],
 
 AS_IF([test x"$enable_debug" != x"no"],
       [dbg=1
+       CFLAGS_save=
+       # Remove -DNDEBUG flag if set
+       for flag in $CFLAGS; do
+           if test "$flag" != "-DNDEBUG"; then
+               CFLAGS_save="$flag $CFLAGS_save"
```

Comment:
I think you might want to make this `CFLAGS_save="$CFLAGS_save $flag"` -- i.e., flip the order so that the original ordering of the flags is preserved.

Reply:
Thanks! This makes sense. The PR has been updated